Fixed argument forwarding in `instance_exec`

dearblue opened the pull request:
However, on C, there is no easy way to pass keyword arguments. Therefore, when called `Kernel#instance_exec` on C, keyword arguments are converted to positional arguments. This is a limitation of current mruby.

fixed #6389

---

During our work we found two problems.

One had to do with commit messages, and calling `mrb_exec_irep()` from C did not pass arguments.
The other was that calls like `instance_exec(1, 2, 3, &:puts)` would result in a `__send__: main is not a symbol nor a string (TypeError)` exception.

I will put these together a bit more and add each as a separate issue.

matz commented:
Thank you! The same change is probably needed for Class#class_exec in mruby-class-ext too.

dearblue commented:
Thank you for the review.
I see, indeed. I will update.

dearblue commented:
Main changes.

  - Fixed argument transfer in `Module#class_exec` as well.
  - Added a common function `mrb_object_exec()` called from `class_exec` and `instance_exec` in `src/vm.c`.

And the argument transfer problem with `mrb_exec_irep()` called from C. It was not difficult to fix.
It is sent as a separate PR.
If that one is taken in first, the branch that calls `mrb_yield_with_classs()` in `mrb_object_exec()` in this PR becomes unnecessary.
